src/File.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace React\Http;
+
+use React\Stream\ReadableStreamInterface;
+
+class File implements FileInterface
+{
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * @var ReadableStreamInterface
+     */
+    protected $stream;
+
+    /**
+     * @param string $name
+     * @param string $filename
+     * @param string $type
+     * @param ReadableStreamInterface $stream
+     */
+    public function __construct($name, $filename, $type, ReadableStreamInterface $stream)
+    {
+        $this->name = $name;
+        $this->filename = $filename;
+        $this->type = $type;
+        $this->stream = $stream;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
```

Comment:
I'm not suggesting we *need* to change the names, merely that this is unclear and needs some documentation :+1: 

That being said, the "name" refers to the (HTML) form field name the file is uploaded as, while the "filename" refers to the actual name of the file. Is the "name" even part of the "file"?